pkg/archive/archive.go

```diff
@@ -552,22 +551,15 @@ func createTarFile(path, extractDir string, hdr *tar.Header, reader io.Reader, L
 		if err != nil {
 			return err
 		}
-		if _, err := io.Copy(file, reader); err != nil {
-			file.Close()
-			return err
-		}
 
-		if err := file.Sync(); err != nil {
-			file.Close()
-			return err
-		}
+		efw := ioutils.NewEagerFileWriter(file)
 
-		if err := unix.Fadvise(int(file.Fd()), 0, 0, unix.FADV_DONTNEED); err != nil {
+		if _, err := io.Copy(efw, reader); err != nil {
 			file.Close()
```

Comment:
Shouldn't this be `efw.Close()` as well?

Reply:
In case of an error there is no point also calling SyncFileRange so I just did `file.Close()` here